gateway/ipfs.go

```diff
@@ -79,23 +80,30 @@ func (g *Gateway) renderIPFSPath(c *gin.Context, base, pth string) {
 				"Links":   links,
 			}
 			c.HTML(http.StatusOK, "/public/html/unixfs.gohtml", params)
-		} else {
-			renderError(c, http.StatusBadRequest, err)
 			return
 		}
-	} else {
-		contentType := http.DetectContentType(data)
-		c.Writer.Header().Set("Content-Type", contentType)
-		c.Render(200, render.Data{Data: data})
+
+		renderError(c, http.StatusBadRequest, err)
+		return
+	}
+	defer f.Close()
+
+	var buf [512]byte
+	n, err := io.ReadAtLeast(f, buf[:], len(buf))
+	if err != nil && err != io.ErrUnexpectedEOF {
+		renderError(c, http.StatusInternalServerError, err)
+		return
 	}
```

Comment:
Try to read at least 512 bytes.
The only possible error that we can come up here is reading less than 512 bytes if the content is less than that size. That's OK.